lib/constructs/interface.js

```diff
@@ -420,25 +420,24 @@ class Interface {
     this.included.push(source);
   }
 
-  generateIterator() {
-    if (!this.iterable) {
+  generateInstallIteratorPrototype() {
+    const { iterable } = this;
+    if (!iterable) {
       return;
     }
 
-    if (this.iterable.isAsync) {
+    if (iterable.isAsync) {
       this.str += `
-        const AsyncIteratorPrototype = Object.create(utils.AsyncIteratorPrototype, {
+        ctorRegistry["${this.name} AsyncIterator"] = Object.assign(Object.create(utils.AsyncIteratorPrototype, {
           [Symbol.toStringTag]: {
             value: "${this.name} AsyncIterator",
             configurable: true
           }
-        });
-
-        Object.assign(AsyncIteratorPrototype, {
+        }, {
           next() {
             const internal = this && this[utils.iterInternalSymbol];
             if (!internal) {
-              return Promise.reject(new TypeError("next() called on a value that is not an async iterator prototype object"));
+              return Promise.reject(new TypeError("next() called on a value that is not a WebIDL default async iterator object"));
```

Comment:
These changes reduce clarity; web developers shouldn't need to know what Web IDL is.